# HG changeset patch
# User Rik <rik@octave.org>
# Date 1502754952 25200
#      Mon Aug 14 16:55:52 2017 -0700
# Node ID adf580507691609c13a2f739efcf001cc8d09cef
# Parent  2e4033e1f28f9c5f6f2a0300c6796320c61529ee
Fix error from TeX parser when 'interpreter' is set to 'none' (bug #51751).

* __axis_label__.m: Set all other property/value pairs (varargin) before
setting "string" property in case they influence the interpratation.

* text.m: Set all other property/value pairs (varargin) before setting "string"
property in case they influence the interpratation.

diff --git a/scripts/plot/appearance/private/__axis_label__.m b/scripts/plot/appearance/private/__axis_label__.m
--- a/scripts/plot/appearance/private/__axis_label__.m
+++ b/scripts/plot/appearance/private/__axis_label__.m
@@ -22,15 +22,16 @@
 ## @end deftypefn
 
 ## Author: jwe
 
 function retval = __axis_label__ (hax, caller, txt, varargin)
 
   h = get (hax, caller);
 
-  set (h, "string", txt, varargin{:});
+  ## Set other prop/values first in case they influence appearance of string.
+  set (h, varargin{:}, "string", txt);
 
   if (nargout > 0)
     retval = h;
   endif
 
 endfunction
diff --git a/scripts/plot/appearance/text.m b/scripts/plot/appearance/text.m
--- a/scripts/plot/appearance/text.m
+++ b/scripts/plot/appearance/text.m
@@ -136,37 +136,35 @@ function h = text (varargin)
   ## Get axis argument which may be in a 'parent' PROP/VAL pair
   [hax, varargin] = __plt_get_axis_arg__ ("text", varargin{:});
   if (isempty (hax))
     hax = gca ();
   else
     hax = hax(1);
   endif
 
-  ## Position argument may alse be in PROP/VAL pair
+  ## Position argument may also be in PROP/VAL pair
   idx = find (strcmpi (varargin, "position"), 1);
   if (idx)
     pos = varargin{idx+1};
     varargin(idx:idx+1) = [];
   else
     pos = [x(:), y(:), z(:)];
   endif
 
   ## Call __go_text__ to do the work
   htmp = zeros (nt, 1);
   if (nx == 1)
-    htmp = __go_text__ (hax, "string", string{1},
-                             ## varargin first, in case "Units" set for pos.
-                             varargin{:},
-                             "position", pos);
+    ## Set varargin first, in case it changes units or interpreter properties.
+    htmp = __go_text__ (hax, varargin{:}, "position", pos,
+                             "string", string{1});
   else
     for n = 1:nt
-      htmp(n) = __go_text__ (hax, "string", string{n},
-                                  varargin{:},
-                                  "position", pos(n,:));
+      htmp(n) = __go_text__ (hax, varargin{:}, "position", pos(n,:),
+                                  "string", string{n});
     endfor
     __request_drawnow__ ();
   endif
 
   if (nargout > 0)
     h = htmp;
   endif
 
